ios/RNCalendarEvents.m

```diff
@@ -640,45 +666,55 @@ - (NSDictionary *)serializeCalendarEvent:(EKEvent *)event
-            [recurrenceRule setValue:@([[rule recurrenceEnd] occurrenceCount]) forKey:@"occurrence"];
+            [formedCalendarEvent setValue:recurrenceRule forKey:_recurrenceRule];
         }
-
-        [formedCalendarEvent setValue:recurrenceRule forKey:_recurrenceRule];
     }
-
+    @catch (NSException *exception) {
+        NSLog(@"RNCalendarEvents encountered an issue will serializing event (recurrenceRules) '%@': %@", event.title, exception.reason);
+    }
+    
     [formedCalendarEvent setValue:[self availabilityStringMatchingConstant:event.availability] forKey:_availability];
     
-    if (event.structuredLocation && event.structuredLocation.title && event.structuredLocation.radius) {
-        NSMutableDictionary *structuredLocation = [[NSMutableDictionary alloc] initWithCapacity:3];
-        [structuredLocation addEntriesFromDictionary: @{
-                                                        @"title": event.structuredLocation.title,
-                                                        @"radius": @(event.structuredLocation.radius)
-                                                        }];
-        if (event.structuredLocation.geoLocation) {
-            [structuredLocation setValue: @{
-                                            @"latitude": @(event.structuredLocation.geoLocation.coordinate.latitude),
-                                            @"longitude": @(event.structuredLocation.geoLocation.coordinate.longitude)
-                                            }
-                                forKey:@"coords"];
+    @try {
+        if (event.structuredLocation && event.structuredLocation.title && event.structuredLocation.radius) {
+            NSMutableDictionary *structuredLocation = [[NSMutableDictionary alloc] initWithCapacity:3];
+            [structuredLocation addEntriesFromDictionary: @{
+                                                            @"title": event.structuredLocation.title,
+                                                            @"radius": @(event.structuredLocation.radius)
+                                                            }];
+            if (event.structuredLocation.geoLocation) {
+                [structuredLocation setValue: @{
+                                                @"latitude": @(event.structuredLocation.geoLocation.coordinate.latitude),
+                                                @"longitude": @(event.structuredLocation.geoLocation.coordinate.longitude)
+                                                }
+                                    forKey:@"coords"];
+            }
+            [formedCalendarEvent setValue: structuredLocation forKey:@"structuredLocation"];
         }
-        [formedCalendarEvent setValue: structuredLocation forKey:@"structuredLocation"];
+    }
+    @catch (NSException *exception) {
+        NSLog(@"RNCalendarEvents encountered an issue will serializing event (structuredLocation) '%@': %@", event.title, exception.reason);
```

Comment:
I appreciate the specific error messaging. I wonder if it would be useful to throw another exception from these blocks so that the downstream rejects can send it to the user?

Reply:
Also, small text change. "will" -> "while" :)

Reply:
Text fixed!

At first I did throw an exception but in practise, it's sucks to get an exception for a fetchAllEvents because a single event have a small issue (or because we are doing something wrong during serialisation).
The exception is a sort of "everything went wrong" so you don't get any result. If a single event is having a small issue (it was the case for me before I found the bug and [fix it](https://github.com/wmcmahan/react-native-calendar-events/commit/1dc570b8980c5a634c7c2c970ffc3218f96cc439)) I think it's better to get most things right (in practise my app become usable instead - otherwise I would have to tell the user "oops something when wrong" because of a small thing with a particular event - which in my case is a value I really don't care about).

Another thing we could do is (instead of throwing), provide an "error" field in the event result, to be able to give the user possibility to precisely know what events have know issue (=>with this lib). I was thinking about doing that later as an improvement.